### PR TITLE
translate: update `langcode()` helper to support Chinese

### PR DESCRIPTION
**Pull request by dgw**
### Description
This is a semi-temporary fix for #2241. It's still not perfect (still allows using unsupported or nonexistent language codes), but there will be additional options available in Sopel 8.0 and beyond. Goal for this is to avoid adding any new dependencies in 7.1.x.

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

File: sopel/modules/translate.py [hunk 1]
```diff
@@ -158,7 +159,16 @@ def tr2(bot, trigger):
...
-        return p.startswith(':') and (2 < len(p) < 10) and p[1:].isalpha()
+        # TODO: We'd be much better off just using the langcodes PyPI package
+        # also TODO: it'd be nice not to require the : prefix, which using
+        # langcodes would make easier in lieu of adding an API key to fetch
+        # the list of supported languages
+        prefixed = p.startswith(':')
+        # the longest codes Google uses (ca. Jan 2022) are zh-CN and zh-TW
+        short_enough = (2 < len(p) < 8)
+        # pesky - in Chinese codes forced a switch from .isalpha(); see #2241
+        fits_pattern = re.match(r':[a-z\-]+', p.lower())
```
Comment (Exirel) [inline sopel/modules/translate.py:170]:
```suggestion
        fits_pattern = re.match(r':[a-z\-]+', p, re.IGNORECASE)
```

**Review by Exirel — APPROVED** [review]
Just a nitpick, doesn't prevent approval.

File: sopel/modules/translate.py (hunk 1 above)
Comment (dgw) [reply sopel/modules/translate.py:170]: I almost did this, tbh. Then I decided it wasn't worth the effort of amending the commit (and making GHA re-test).

Comment (Exirel) [reply sopel/modules/translate.py:170]: Did you try to compile the regex first? 😁

Comment (dgw) [reply sopel/modules/translate.py:170]: I thought it was memoized, but maybe `timeit` skips over that built-in optimization.

```pycon
>>> timeit('r.match(s.lower())', 'import re; s = ":zh-CN"; r = re.compile(r":[a-z\-]+")')
0.19142690021544695
>>> timeit('r.match(s)', 'import re; s = ":zh-CN"; r = re.compile(r":[a-z\-]+", re.IGNORECASE)')
0.2070107003673911
```

So it's _really_ a "who cares?" moment, then. 😂

**Pull request merged**